apps/web/src/app/[locale]/(logged-in)/aside.tsx

```diff
@@ -1,22 +1,57 @@
 'use client';
 
 import { Layers, MenuIcon, PanelLeftClose, PanelLeftOpen, Settings, X } from 'lucide-react';
-import { useState } from 'react';
+import { useCallback, useEffect, useState } from 'react';
 import { cx } from '@repo/ui/tailwind-utils';
 import Link from 'next/link';
+import { usePathname, useRouter, useSearchParams } from 'next/navigation';
 import { SignOutBtn } from '@/components/login/sign-out-btn';
 import { ActiveLink, type LinkType } from '@/components/home/active-link';
 import { LogoFull } from '@/app/[locale]/(logged-in)/logo-full';
+import { InsightsColumnsGroupBy } from '@/graphql/generated/schema-server';
+import { createURLWithQueryParams, type QueryParamsType } from '@/app/[locale]/(logged-in)/insights/query-string-util';
 
 const links: LinkType[] = [
   { url: '/insights', text: 'Insights', icon: <Layers /> },
   { url: '/settings/integrations', text: 'Settings', icon: <Settings /> },
 ];
 
 export function Aside({ children }: { children: React.ReactNode }): React.ReactNode | null {
+  const router = useRouter();
+  const pathname = usePathname();
+  const searchParams = useSearchParams();
   const [open, setOpen] = useState(false);
   const [minimize, setMinimize] = useState(false);
 
+  const setInitialGroupedByFilters = useCallback(() => {
+    const initialParams: QueryParamsType = [
+      {
+        key: 'groupedBy',
+        value: InsightsColumnsGroupBy.adId,
+      },
+      {
+        key: 'groupedBy',
+        value: InsightsColumnsGroupBy.device,
+      },
+      {
+        key: 'groupedBy',
+        value: InsightsColumnsGroupBy.publisher,
+      },
+      {
+        key: 'groupedBy',
+        value: InsightsColumnsGroupBy.position,
+      },
+    ];
+    const newURL = createURLWithQueryParams(pathname, initialParams);
+    router.replace(newURL);
+  }, [pathname, router]);
+
+  useEffect(() => {
+    if (pathname === '/insights') {
```

Comment:
```suggestion
     if (pathname.endsWith('/insights') && searchParams.size === 0) {
```
locale break the `===` also we want to make sure that it does not have any search params.
The problem here is that if for any reason the user removes all searchparams it will reset. We may add a custom `dummy=true` searchparam, but I don't like it a lot

Reply:
Nice catch!

Reply:
don't forget the `endsWith` if someone is using locale just `===` does not work.

Regarding the not reset to defaults on empty, I was thinking that we can add in `query-string-util.ts` after line 64
```
        if (params.getAll(key).length === 0) {
          params.set('g', 'none');
        }
```

Reply:
or something similar